src/script/browser/CheckBrowser.ts

```diff
@@ -17,119 +17,86 @@
+    } else {
+      if (navigator.userAgent.toLowerCase().indexOf('firefox') !== -1) {
+        let dbOpenRequest: IDBOpenDBRequest;
 
-      try {
-        dbOpenRequest = window.indexedDB.open('test');
-      } catch (error) {
-        return Promise.reject(new Error('Error initializing IndexedDB'));
-      }
+        try {
+          dbOpenRequest = window.indexedDB.open('test');
+        } catch (error) {
+          return resolve(false);
+        }
 
-      return new Promise((resolve, reject) => {
-        const connectionTimeout = setTimeout(
-          () => reject(new Error('Error opening IndexedDB (response timeout)')),
-          10000,
-        );
+        const connectionTimeout = setTimeout(() => resolve(false), 10000);
         dbOpenRequest.onerror = event => {
           clearTimeout(connectionTimeout);
           if (dbOpenRequest.error) {
             event.preventDefault();
-            return reject(new Error('Error opening IndexedDB'));
+            return resolve(false);
           }
           return undefined;
         };
         dbOpenRequest.onsuccess = event => {
           clearTimeout(connectionTimeout);
-          resolve();
+          resolve(true);
         };
-      });
+      }
+      resolve(true);
     }
+  }).catch(() => false);
 
-    return Promise.resolve();
-  };*/
-
-const isBrowserSupported = (): boolean =>
-  (!isMobile() || isOauth()) &&
-  supportsIndexDB() &&
-  supportsCookies() &&
-  'Promise' in window &&
-  'allSettled' in Promise &&
-  'Symbol' in window &&
-  'replace' in Symbol &&
-  'WeakMap' in window &&
-  'assign' in Object &&
-  'entries' in Object &&
-  'values' in Object &&
-  'getOwnPropertyDescriptors' in Object &&
-  'fromEntries' in Object &&
-  'assign' in Object &&
-  'URL' in window &&
-  'toJSON' in URL.prototype &&
-  'URLSearchParams' in window &&
-  Array.prototype[Symbol.iterator] &&
-  'includes' in Array.prototype &&
-  'reduce' in Array.prototype &&
-  'sort' in Array.prototype &&
-  'flatMap' in Array.prototype &&
-  NodeList.prototype[Symbol.iterator] &&
-  'forEach' in NodeList.prototype &&
-  HTMLCollection.prototype[Symbol.iterator] &&
-  DOMTokenList.prototype[Symbol.iterator] &&
-  'forEach' in DOMTokenList.prototype &&
-  'fill' in Int8Array.prototype &&
-  'set' in Int8Array.prototype &&
-  'sort' in Int8Array.prototype &&
-  'replace' in String.prototype &&
-  'search' in String.prototype &&
-  'split' in String.prototype &&
-  'includes' in String.prototype &&
-  'match' in String.prototype &&
-  'trim' in String.prototype &&
-  'split' in String.prototype &&
-  'endsWith' in String.prototype &&
-  'replaceAll' in String.prototype &&
-  'sticky' in RegExp.prototype &&
-  'toString' in RegExp &&
-  parseFloat('1.23') === 1.23 &&
-  (1.23456789).toFixed(2) === '1.23' &&
-  'RTCPeerConnection' in window;
+const checkBrowser = (): void => {
+  if (isOauth()) {
+    return;
+  }
+  if (!('RTCPeerConnection' in window)) {
+    location.href = '/unsupported/';
+  }
```

Comment:
why is this one here and not in `browser.js`?

Reply:
I was just thinking of leaving in `browser.js` only the most primitive things. At first, I even thought to put there only `globalThis`, `Promise`, `Object`, `Array`. My logic was like... what if we identify similar issues with `RTC` support, as with `IndexDB`? We are forced to have 2 separate files anyway, so I was dividing the logic somehow intuitively - basic primitive things into the static JS files, some more advanced code - into the TS file. Let me know if you think it's better to move it there

Reply:
Ok, it makes sense I think :)

Maybe, since this one is a proper ts file, maybe we can 'import' it from the source files (rather than from html) WDYT?

Reply:
I'm not sure the browser won't get blocked by some unsupported code inside, for instance, `vendor.js`, if we import it in the main code. It might work, as the base features are already detected within `browser.js`, so the chances are already pretty low. But we wanted to completely move it out of the main codebase initially for this reason. Right now the check browser scripts run instantly in the `<head>` of the HTMLs. Also, they are very readable this way - you see them both imported one by one in the HTML. WDYT?

Reply:
yeah, it makes sense to do the rtc check later in the app. (and only for `app.ts`, not the auth page). 

But since those are application specific checks, they do belong in the app code I believe. It would make sense to import this file from `src/script/main/index.tsx` I believe

Reply:
I'm not sure about not adding it to the auth page: if a user logs in and then immediately gets the page "your browser is not supported" - then what is the point of allowing the user to log in at all?